source/constructs/api/common/query_condition.py

```diff
@@ -22,6 +23,60 @@
     group_column: Optional[str] = const.EMPTY_STR
 
 
+def query_with_func_condition(query: Query, condition: QueryCondition,
+                              sum_aggregate_columns: [] = None, count_aggregate_columns: [] = None):
+    table_obj = query.column_descriptions[0]['type']
+    free_search_columns = []
+    free_search_conditions = []
+    in_conditions = []
+    if sum_aggregate_columns:
+        for sum_aggregate_column in sum_aggregate_columns:
+            aggregate_column_obj = getattr(table_obj, sum_aggregate_column)
+            query = query.add_columns(func.sum(aggregate_column_obj).label(f"{sum_aggregate_column}_sum"))
+    if count_aggregate_columns:
+        for count_aggregate_column in count_aggregate_columns:
+            aggregate_column_obj = getattr(table_obj, count_aggregate_column)
+            query = query.add_columns(func.sum(aggregate_column_obj).label(f"{count_aggregate_column}_count"))
+
+    for column in [item for item in dir(table_obj) if not item.startswith('_')]:
+        if hasattr(getattr(table_obj, column), 'info') and getattr(table_obj, column).info.get('searchable'):
+            free_search_columns.append(column)
+    if condition and condition.conditions:
+        for c in condition.conditions:
+            if c.column:
+                table_value = getattr(table_obj, c.column)
+                if c.operation == OperationType.EQUAL.value:
+                    query = query.filter(table_value == c.values)
+                elif c.operation == OperationType.NOT_EQUAL.value:
+                    query = query.filter(table_value != c.values)
+                elif c.operation == OperationType.CONTAIN.value:
+                    query = query.filter(table_value.contains(c.values))
+                elif c.operation == OperationType.IN.value:
+                    for item in c.values:
+                        in_conditions.append(table_value == item)
+                    query = query.filter(or_(*in_conditions))
+                else:
+                    query = query.filter(or_(table_value == None, not_(table_value.contains(c.values))))
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The `==` and `!=` operators use the compared objects&#39; ` __eq__` method to test if they are equal. To check if an object is a singleton, such as `None`, we recommend that you use the `is` identity comparison operator.

[Learn more](https://docs.python.org/3/reference/expressions.html#is)